devlab/tests/test_group_verification.py

```diff
@@ -0,0 +1,167 @@
+    """
+    Testing class for grouping verification.
+    """
+    def setUp(self):
+        """
+        SetUp method.
+        """
+        self.src_cloud = Prerequisites(cloud_prefix='SRC')
+        self.main_folder = os.path.dirname(os.path.dirname(os.getcwd()))
+        # TODO:
+        #  Using relative paths is a bad practice, unfortunately this is the
+        #  only way at this moment.
+        #  Should be fixed by implementing proper package module for Cloud Ferry.
+        cmd_no_path = './../../devlab/provision/generate_config.sh --cloudferry-path '
+        cmd = os.path.join(cmd_no_path, self.main_folder.lstrip('/'))
+        os.system(cmd)
+
+    def tearDown(self):
+        """
+        Clean up method.
+        """
+        cmd_1 = 'rm {}'.format(os.path.join(self.main_folder, 'devlab/tests', self.new_file_name.lstrip('/')))
+        cmd_2 = 'rm {}'.format(os.path.join(self.main_folder, 'vm_groups.yaml'))
+        for cmd in [cmd_1, cmd_2]:
+            try:
+                os.system(cmd)
+            except Exception as e:
+                print 'Was unable to delete testing files, error output:' \
+                      '\n{}'.format(e)
+
+    def _prepare_files(self, grouping_by):
+        """
+        Method for generation of test/configuration files.
+        """
+        self.post_conf_dict = {}
+        self.pre_conf_dict = {}
+        main_folder = self.main_folder
+
+        file_path = 'devlab/tests/groups_example.yaml'
+        exmpl_file_path = os.path.join(main_folder, file_path)
+        pre_conf = open(exmpl_file_path, 'r')
+        self.pre_conf_dict = yaml.load(pre_conf)
+
+        inst_id_list = []
+        inst_3 = None
+        for key in self.pre_conf_dict.keys():
+            if key == 'user_defined_group_1':
+                for val in self.pre_conf_dict[key]:
+                    for inst in self.src_vms:
+                        if inst['name'] == val:
+                            inst_id_list.append(inst['id'])
+            elif key == 'user_defined_group_2':
+                for inst in self.src_vms:
+                    if inst['name'] == self.pre_conf_dict[key][0]:
+                        inst_3 = inst['id']
+        self.pre_conf_dict['group_by'] = [unicode(grouping_by)]
+        self.pre_conf_dict['user_defined_group_1'] = inst_id_list
+        self.pre_conf_dict['user_defined_group_2'] = [inst_3]
+        self.new_file_name = 'test_file.yaml'
+        conf_file = open(os.path.join(os.getcwd(), self.new_file_name), 'w')
+        conf_file.write(yaml.dump(self.pre_conf_dict, default_flow_style=False))
+        conf_file.close()
+        fab_path = os.path.join('devlab/tests', self.new_file_name)
+        cmd = 'cd {} && fab get_groups:configuration.ini,{}'.format(main_folder, fab_path)
+        os.system(cmd)
+        post_file_path = os.path.join(main_folder, 'vm_groups.yaml')
+        post_conf = open(post_file_path, 'r')
+        self.post_conf_dict = yaml.load(post_conf)
```

Comment:
`yaml.load` requires stream descriptor, would be better to have `yaml.load(file(<filename>, 'r'))`